### PR TITLE
NDS-127 Prev & Next Component

### PR DESCRIPTION
**Pull request by wa-rren-dev**
https://nicedigital.atlassian.net/browse/NDS-127

### CODE REVIEW

File: components/nds-prev-next/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+<PrevNext
+	nextPageLink={{
+		text: "About us",
+		destination: "/about-us",
+		intro: "Find out more"
+	}}
+/>
+```
+
+##### classname
```
Comment (johnholland-nice) [inline components/nds-prev-next/README.md:99]: Should this be camel case? Sorry if that's pedantic.

File: components/nds-prev-next/__tests__/nds-prev-next.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		expect(toJson(wrapper)).toMatchSnapshot();
+	});
+
+	it("link should render an href attribute if the elementType is an anchor", () => {
+		const wrapper = mount(<PrevNext previousPageLink={previousPageLink} />);
+		const anchor = wrapper.find("a");
+		expect(anchor.props()["href"]).toEqual("/alert");
+	});
+
+	it("link should render an href attribute if the elementType is an anchor", () => {
```
Comment (johnholland-nice) [inline components/nds-prev-next/__tests__/nds-prev-next.test.js:46]: Duplicated test

File: components/nds-prev-next/__tests__/nds-prev-next.test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		const wrapper = mount(<PrevNext previousPageLink={previousPageLink} />);
+		const anchor = wrapper.find("a");
+		expect(anchor.props()["href"]).toEqual("/alert");
+	});
```
Comment (johnholland-nice) [inline components/nds-prev-next/__tests__/nds-prev-next.test.js:50]: Duplicated test

**Review by johnholland-nice — DISMISSED** [review]
Added a couple of comments on things I spotted.

File: components/nds-prev-next/nds-prev-next.d.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+	interface PrevNextLink {
+		text?: string;
+		destination: string;
+		elementType?: React.ElementType;
+		intro?: string;
+	}
+
+	interface PrevNextProps {
+		previousPageLink: PrevNextLink;
```
Comment (ediblecode) [inline components/nds-prev-next/nds-prev-next.d.ts:12]: Do both of these need to be optional?

File: components/nds-prev-next/src/PrevNext.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	};
+	if (LinkTag === "a") {
+		linkProps.href = destination;
+	} else {
+		linkProps.to = destination;
+	}
+	return (
+		<LinkTag {...linkProps}>
+			<span className="prev-next__link-intro">{intro}</span>
+			<span className="prev-next__link-text">{text}</span>
```
Comment (ediblecode) [inline components/nds-prev-next/src/PrevNext.js:18]: The `text` prop in the TS def is optioanl, in which case should we conditionally render this span?

File: components/nds-prev-next/src/PrevNext.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		</LinkTag>
+	);
+};
+
+export const PrevNext = props => {
+	const { nextPageLink, previousPageLink, className, ...rest } = props;
+
+	const nextLinkProps = {
+		intro: nextPageLink?.intro || "Next page",
+		text: nextPageLink?.text || nextPageLink?.destination,
```
Comment (ediblecode) [inline components/nds-prev-next/src/PrevNext.js:28]: Not sure we should be using the destination as the fallback for link text? Isn't it better to just have the intro without text?

File: packages/docs/src/content/components/previous-next/index.mdx [hunk 7]
```diff
@@ -1,17 +1,33 @@
 ---
 title: Previous and Next
 navigationLabel: Previous & Next
-description: A way for users to navigation into previous or next pages of content.
+description: A way for users to navigate to previous or next pages in a sequence of content.
+inpagenav: true
 template: detail
+storybook: ?path=/story/components-previous-next
```
Comment (ediblecode) [inline packages/docs/src/content/components/previous-next/index.mdx:7]: The `storybook` prop for action-banner is just the component name, and not the full path - should they be consistent?

File: packages/storybook/stories/nds-tag.stories.js [hunk 8]
```diff
@@ -5,7 +5,7 @@ import { text, select } from "@storybook/addon-knobs";
 
 import { Tag } from "@nice-digital/nds-tag";
 
-storiesOf("Tag", module).add("Try it out", () => {
+storiesOf("Foundations/Tag", module).add("Try it out", () => {
```
Comment (ediblecode) [inline packages/storybook/stories/nds-tag.stories.js:8]: Isn't tag a component not a foundation?

**Review by ediblecode — CHANGES_REQUESTED** [review]

File: packages/docs/src/content/components/previous-next/index.mdx (hunk 7 above)
Comment (wa-rren-dev) [reply packages/docs/src/content/components/previous-next/index.mdx:7]: yep - sorting that on action-banner branch now

File: components/nds-prev-next/src/PrevNext.js (hunk 5 above)
Comment (wa-rren-dev) [reply components/nds-prev-next/src/PrevNext.js:18]: There'll always need to be text of some kind - at this point the text is at least the destination

**Pull request merged**